src/lib/network_pool/snark_pool.ml

```diff
@@ -168,11 +168,15 @@ end)
                         return ()
                       else (
                         removedCounter := 0 ;
+                        Statement_table.filter_keys_inplace t.snark_table
+                          ~f:(fun work ->
+                            Option.is_some
+                              (Statement_table.find refcount_table work) ) ;
                         return
-                          (Statement_table.filter_keys_inplace t.snark_table
-                             ~f:(fun work ->
-                               Option.is_some
-                                 (Statement_table.find refcount_table work) )) )
```

Comment:
you can't just not clear things from the table anymore

Reply:
What do you mean? I am still clearing it

Reply:
oh, I'm blind